### PR TITLE
[RN] Remove isMounted() false positive warning

### PR DESCRIPTION
**Pull request by gaearon**
Fixes https://github.com/facebook/react-native/issues/18868#issuecomment-413579095.

This removes a noisy false positive warning that occurs on RN whenever you call something like `measureInWindow()` on a `NativeComponent` descendant. The warning occurred because `mountSafeCallback` checks for `isMounted` (and calls if it exists), but `NativeComponent` only inherits an `isMounted` getter that warns about its deprecation from `React.Component`.

In fact `isMounted()` wasn't being called for host components as the original code probably assumed — because they [don't have](https://github.com/facebook/react/blob/5031ebf6beddf88cac15f4d2c9e91f8dbb91d59d/packages/react-native-renderer/src/ReactNativeFiberHostComponent.js#L33-L98) a method called like this.

So the method was already lying about its usefulness. It only works on old-style `createClass()` components. Which happen to define a field called `__isMounted`. So I switched the method to check for that field alone, and reflected its true nature in its naming.

**TLDR: this removes a noisy false positive warning, and leaves a TODO to fix the method because it doesn't do what it claims.**

### CODE REVIEW

File: packages/react-native-renderer/src/NativeMethodsMixinUtils.js [hunk 1]
```diff
@@ -11,26 +11,36 @@
...
-      // It is currently necessary because View uses createClass,
-      // and so any measure() calls on View (which are done by React
-      // DevTools) trigger the isMounted() deprecation warning.
       if (!context.__isMounted) {
         return undefined;
       }
-      // The else branch is important so that we don't
-      // trigger the deprecation warning by calling isMounted.
-    } else if (typeof context.isMounted === 'function') {
-      if (!context.isMounted()) {
```
Comment (gaearon) [inline packages/react-native-renderer/src/NativeMethodsMixinUtils.js]: My hypothesis is we never actually hit this.

ES6 class components always return `undefined` from `isMounted` and log a warning.

RN host components don't have a method called `isMounted()`.

I guess a *user* could define their own `isMounted()` implementation but it wouldn't make sense to intentionally support this.

**Comment by react-linter** [comment]
<details>
  <summary>Details of bundled changes.</summary>

  <p>Comparing: 1c0ba70b4b753178d1bb9d9d8e32b90c50ec7e2c...19c014934195bb918ea0839d76f33caabf2f80ac</p>


  
## react-native-renderer
File | Filesize Diff | Gzip Diff | Prev Size | Current Size | Prev Gzip | Current Gzip | ENV
 ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  --- 
ReactNativeRenderer-dev.js | +0.1% | +0.2% | 497.24 KB | 497.86 KB | 109.95 KB | 110.15 KB | RN_FB_DEV
**ReactNativeRenderer-prod.js** | **0.0%** | **0.0%** | **214.7 KB** | **214.79 KB** | **37.47 KB** | **37.47 KB** | **RN_FB_PROD**
ReactNativeRenderer-dev.js | +0.1% | +0.2% | 496.98 KB | 497.59 KB | 109.89 KB | 110.09 KB | RN_OSS_DEV
**ReactNativeRenderer-prod.js** | **0.0%** | **0.0%** | **204.52 KB** | **204.61 KB** | **35.82 KB** | **35.82 KB** | **RN_OSS_PROD**
ReactFabric-dev.js | +0.1% | +0.2% | 487.43 KB | 488.04 KB | 107.54 KB | 107.73 KB | RN_FB_DEV
**ReactFabric-prod.js** | **0.0%** | **0.0%** | **197.03 KB** | **197.11 KB** | **34.33 KB** | **34.33 KB** | **RN_FB_PROD**
ReactFabric-dev.js | +0.1% | +0.2% | 487.46 KB | 488.08 KB | 107.55 KB | 107.75 KB | RN_OSS_DEV
**ReactFabric-prod.js** | **0.0%** | **0.0%** | **197.06 KB** | **197.15 KB** | **34.34 KB** | **34.34 KB** | **RN_OSS_PROD**
ReactNativeRenderer-profiling.js | 0.0% | -0.0% | 207.94 KB | 208.03 KB | 36.49 KB | 36.49 KB | RN_OSS_PROFILING
ReactFabric-profiling.js | 0.0% | 0.0% | 200.03 KB | 200.12 KB | 34.9 KB | 34.9 KB | RN_OSS_PROFILING
ReactNativeRenderer-profiling.js | 0.0% | 0.0% | 218.1 KB | 218.19 KB | 38.12 KB | 38.12 KB | RN_FB_PROFILING
ReactFabric-profiling.js | 0.0% | 0.0% | 199.99 KB | 200.08 KB | 34.88 KB | 34.88 KB | RN_FB_PROFILING

  </details>
  
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: packages/react-native-renderer/src/NativeMethodsMixin.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ import UIManager from 'UIManager';
 
 import * as ReactNativeAttributePayload from './ReactNativeAttributePayload';
 import {
-  mountSafeCallback,
+  mountSafeCallback_NOT_REALLY_SAFE,
```
Comment (bvaughn) [inline packages/react-native-renderer/src/NativeMethodsMixin.js]: "safe (not really safe)" 😆 😭

**Comment by bvaughn** [comment]
Unfortunately I don't because I'm currently working from a broken temporary laptop.

**Comment by gaearon** [comment]
Meh. I think this is correct

File: packages/react-native-renderer/src/NativeMethodsMixin.js (hunk 2 above)
Comment (gaearon) [reply packages/react-native-renderer/src/NativeMethodsMixin.js]: Truth hurts
